Adding more logs and traces

haverma opened the pull request:
1.) Adding more information in logging (mostly container name)
2.) Adding traces for computeDataplane operation
3.) Adding trace ID to all logs if tracing is enabled (Had to use the uber specific SpanContext implementation for this as OpenTrace specification does not expose trace-id getters), but they have an active [issue](https://github.com/opentracing/specification/issues/24) open for this in OpenTrace.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/batfish/batfish/826)
<!-- Reviewable:end -->

dhalperi commented:
On track.

---

Reviewed 10 of 10 files at r1.
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[projects/batfish-common-protocol/src/main/java/org/batfish/common/BatfishLogger.java, line 376 at r1](https://reviewable.io:443/reviews/batfish/batfish/826#-L2JiqbUbGf2PXHMZQRY:-L2JiqbUbGf2PXHMZQRZ:b-bmm9sw) ([raw file](https://github.com/batfish/batfish/blob/7863964d95af6c41f976a3826199d8245955f31b/projects/batfish-common-protocol/src/main/java/org/batfish/common/BatfishLogger.java#L376)):*
> ```Java
>         outputMsg =
>             String.format(
>                 "%s; trace-id:%s\n", outputMsg.trim(), Long.toHexString(context.getTraceId()));
> ```

We're going to have to figure something else out here. This makes us unable to use a different tracing backend.

---

*[projects/batfish/src/main/java/org/batfish/bdp/BdpDataPlanePlugin.java, line 71 at r1](https://reviewable.io:443/reviews/batfish/batfish/826#-L2Jj8a9n4LHBpUrCwlB:-L2Jj8a9n4LHBpUrCwlC:b-9lni5a) ([raw file](https://github.com/batfish/batfish/blob/7863964d95af6c41f976a3826199d8245955f31b/projects/batfish/src/main/java/org/batfish/bdp/BdpDataPlanePlugin.java#L71)):*
> ```Java
>     _batfish.newBatch("Writing data plane to disk", 0);
>     try (ActiveSpan writeDataplane =
>         GlobalTracer.get().buildSpan("Writing data-plane").startActive()) {
> ```

Is it a `data-plane` or a `data plane`? Use consistent language.

---

*[projects/batfish/src/main/java/org/batfish/bdp/BdpEngine.java, line 488 at r1](https://reviewable.io:443/reviews/batfish/batfish/826#-L2JjNS9mRdoz14qrudC:-L2JjNS9mRdoz14qrudD:bsq6qr6) ([raw file](https://github.com/batfish/batfish/blob/7863964d95af6c41f976a3826199d8245955f31b/projects/batfish/src/main/java/org/batfish/bdp/BdpEngine.java#L488)):*
> ```Java
> 
>       try (ActiveSpan activeStaticRoutesSpan =
>           GlobalTracer.get().buildSpan("Activate Static Routes").startActive()) {
> ```

Use consistent capitalization scheme: `dependent routes` vs `Static Routes`?

---

*[projects/batfish/src/main/java/org/batfish/main/Batfish.java, line 1017 at r1](https://reviewable.io:443/reviews/batfish/batfish/826#-L2Jjc9tcKCSeLUKODE0:-L2Jjc9tcKCSeLUKODE1:bysdjkz) ([raw file](https://github.com/batfish/batfish/blob/7863964d95af6c41f976a3826199d8245955f31b/projects/batfish/src/main/java/org/batfish/main/Batfish.java#L1017)):*
> _settings.getContainerDir().getFileName().toString()

why doesn't this use `#getContainerName()`?

---

*[projects/batfish/src/main/java/org/batfish/main/Driver.java, line 362 at r1](https://reviewable.io:443/reviews/batfish/batfish/826#-L2JjmYlgLIrr8cqLaX0:-L2JjmYmUeabxwOBoYk8:b4pfrhi) ([raw file](https://github.com/batfish/batfish/blob/7863964d95af6c41f976a3826199d8245955f31b/projects/batfish/src/main/java/org/batfish/main/Driver.java#L362)):*
> ```Java
>               try (ActiveSpan runBatfishSpan =
>                   GlobalTracer.get()
>                       .buildSpan("Run Batfish")
> ```

Improve the name of this span? What is it actually?

---

*[projects/batfish/src/main/java/org/batfish/main/Driver.java, line 488 at r1](https://reviewable.io:443/reviews/batfish/batfish/826#-L2JjsK2118-aW3oXE5W:-L2JjsK2118-aW3oXE5X:byg71dg) ([raw file](https://github.com/batfish/batfish/blob/7863964d95af6c41f976a3826199d8245955f31b/projects/batfish/src/main/java/org/batfish/main/Driver.java#L488)):*
> ```Java
>                   try (ActiveSpan runBatfishSpan =
>                       GlobalTracer.get()
>                           .buildSpan("Run Batfish through service")
> ```

better name for this span?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/batfish/batfish/826#-:-L2Jk0MsfGgiAhTcNdyF:b-sn6ktd)*
<!-- Sent from Reviewable.io -->

haverma commented:
Addressed comments

---

Review status: 2 of 8 files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[projects/batfish/src/main/java/org/batfish/bdp/BdpDataPlanePlugin.java, line 71 at r1](https://reviewable.io:443/reviews/batfish/batfish/826#-L2Jj8a9n4LHBpUrCwlB:-L2NyAENTbrZ43nSOyX6:bb6un1j) ([raw file](https://github.com/batfish/batfish/blob/7863964d95af6c41f976a3826199d8245955f31b/projects/batfish/src/main/java/org/batfish/bdp/BdpDataPlanePlugin.java#L71)):*
<details><summary><i>Previously, dhalperi (Dan Halperin) wrote…</i></summary><blockquote>

Is it a `data-plane` or a `data plane`? Use consistent language.
</blockquote></details>

Done

---

*[projects/batfish/src/main/java/org/batfish/bdp/BdpEngine.java, line 488 at r1](https://reviewable.io:443/reviews/batfish/batfish/826#-L2JjNS9mRdoz14qrudC:-L2NyBn5YXPk3ntZZg6A:bb6un1j) ([raw file](https://github.com/batfish/batfish/blob/7863964d95af6c41f976a3826199d8245955f31b/projects/batfish/src/main/java/org/batfish/bdp/BdpEngine.java#L488)):*
<details><summary><i>Previously, dhalperi (Dan Halperin) wrote…</i></summary><blockquote>

Use consistent capitalization scheme: `dependent routes` vs `Static Routes`?
</blockquote></details>

Done

---

*[projects/batfish/src/main/java/org/batfish/main/Batfish.java, line 1017 at r1](https://reviewable.io:443/reviews/batfish/batfish/826#-L2Jjc9tcKCSeLUKODE0:-L2NyCaSQAVmmURndsNJ:bb6un1j) ([raw file](https://github.com/batfish/batfish/blob/7863964d95af6c41f976a3826199d8245955f31b/projects/batfish/src/main/java/org/batfish/main/Batfish.java#L1017)):*
<details><summary><i>Previously, dhalperi (Dan Halperin) wrote…</i></summary><blockquote>

> _settings.getContainerDir().getFileName().toString()

why doesn't this use `#getContainerName()`?
</blockquote></details>

Done

---

*[projects/batfish/src/main/java/org/batfish/main/Driver.java, line 362 at r1](https://reviewable.io:443/reviews/batfish/batfish/826#-L2JjmYlgLIrr8cqLaX0:-L2NyDYQxs5eBDoGMMB_:bb6un1j) ([raw file](https://github.com/batfish/batfish/blob/7863964d95af6c41f976a3826199d8245955f31b/projects/batfish/src/main/java/org/batfish/main/Driver.java#L362)):*
<details><summary><i>Previously, dhalperi (Dan Halperin) wrote…</i></summary><blockquote>

Improve the name of this span? What is it actually?
</blockquote></details>

Done

---

*[projects/batfish/src/main/java/org/batfish/main/Driver.java, line 488 at r1](https://reviewable.io:443/reviews/batfish/batfish/826#-L2JjsK2118-aW3oXE5W:-L2NyFWBb7UQFJrdHONs:bb6un1j) ([raw file](https://github.com/batfish/batfish/blob/7863964d95af6c41f976a3826199d8245955f31b/projects/batfish/src/main/java/org/batfish/main/Driver.java#L488)):*
<details><summary><i>Previously, dhalperi (Dan Halperin) wrote…</i></summary><blockquote>

better name for this span?
</blockquote></details>

Done

---

*[projects/batfish-common-protocol/src/main/java/org/batfish/common/BatfishLogger.java, line 376 at r1](https://reviewable.io:443/reviews/batfish/batfish/826#-L2JiqbUbGf2PXHMZQRY:-L2NyGs1uudSM0ODu507:b85t9dn) ([raw file](https://github.com/batfish/batfish/blob/7863964d95af6c41f976a3826199d8245955f31b/projects/batfish-common-protocol/src/main/java/org/batfish/common/BatfishLogger.java#L376)):*
<details><summary><i>Previously, dhalperi (Dan Halperin) wrote…</i></summary><blockquote>

We're going to have to figure something else out here. This makes us unable to use a different tracing backend.
</blockquote></details>

removed, now using work item id instead for correlation

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/batfish/batfish/826#-:-L2NyNg8Yacgpbd_W_ow:b-d9w856)*
<!-- Sent from Reviewable.io -->

dhalperi commented:
Reviewed 5 of 8 files at r2, 4 of 4 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/batfish/batfish/826)*
<!-- Sent from Reviewable.io -->
